rundeckapp/src/test/groovy/rundeck/controllers/ScheduledExecutionControllerSpec.groovy

```diff
@@ -2944,4 +2944,79 @@ class ScheduledExecutionControllerSpec extends HibernateSpec implements Controll
         model.scheduledExecution != null
         job.options[0].valuesFromPlugin == null
     }
+
+    @Unroll
+    def "test edit Job"() {
+        ScheduledExecution.metaClass.static.withNewSession = {Closure c -> c.call() }
+
+        ScheduledExecution job = new ScheduledExecution(createJobParams())
+        job.addToOptions(new Option(name: 'optvals', optionValuesPluginType: 'test', required: true, enforced: false))
+        job.save()
+        controller.scheduledExecutionService = Mock(ScheduledExecutionService){
+            getByIDorUUID(_) >> job
+        }
+        def auth = Mock(UserAndRolesAuthContext){
+            getUsername() >> 'bob'
+        }
+
+        NodeSetImpl testNodeSetB = new NodeSetImpl()
+        testNodeSetB.putNode(new NodeEntryImpl("nodea"))
+
+        controller.frameworkService = Mock(FrameworkService){
+            getAuthContextForSubjectAndProject(*_) >> auth
+            authorizeProjectJobAll(_, job, ['update','read'], 'AProject') >> true
+            filterAuthorizedNodes(_,_,_,_)>>{args-> args[2]}
+            filterNodeSet({ NodesSelector selector->
+                selector.acceptNode(new NodeEntryImpl("nodea")) &&
+                        selector.acceptNode(new NodeEntryImpl("nodec xyz")) &&
+                        !selector.acceptNode(new NodeEntryImpl("nodeb"))
+
+            },_)>>testNodeSetB
+            getRundeckFramework()>>Mock(Framework){
+                getFrameworkNodeName()>>'fwnode'
+            }
+            getProjectGlobals(_) >> [:]
+        }
+
+        controller.apiService = Mock(ApiService)
+        controller.rundeckAuthorizedServicesProvider = Mock(AuthorizedServicesProvider)
+        controller.optionValuesService = Mock(OptionValuesService){
+            getOptions('AProject', 'test',_) >> {throw new Exception ("error calling plugin")}
+        }
+
+        def notificationPlugins = ["notification-plugin": ["value1": "label1"]]
+
+        controller.notificationService = Mock(NotificationService) {
+            1 * listNotificationPlugins() >> [:]
+            1 * listNotificationPluginsDynamicProperties("AProject", _) >> notificationPlugins
```

Comment:
was the error caused when an exception was thrown in the listNotificationPluginsDynamicProperties method? should the test reflect that?

Reply:
yes you right, but the problem didn't happen because listNotificationPluginsDynamicProperties returned an exception, it happened when an exception was triggered inside that method (but handled ). Can I reproduce that in the test?